### PR TITLE
add missing entities

### PR DESCRIPTION
**Pull request by sevgibson**
FYI, I realize this looks huge -- the first 5 files are the substance. Most of the files are specs / cassettes.

I need a couple of these endpoints for https://github.com/PipelineDeals/pipeline_integrations/pull/216

Adding all other documented endpoints and specs, while I'm here.

Note: I didn't add team specs because I am getting quota errors locally -- but other than teams, all the admin endpoints are at least being hit for create, index, destroy.

### CODE REVIEW

File: lib/pipeline/admin/feature.rb [hunk 1]
```diff
@@ -6,10 +6,10 @@ def initialize(pipeline:, id: nil, attributes: nil)
   end
 
   def remove_provider(provided_by)
-    _delete("features/#{id}", query: { provided_by: })
+    _delete("features/#{id}", query: { provided_by: provided_by })
```
Comment (sevgibson) [inline lib/pipeline/admin/feature.rb:9]: backward compatible ruby

File: lib/pipeline.rb [hunk 2]
```diff
@@ -67,6 +71,10 @@ def documents
     Pipeline::Documents.new(pipeline: self)
   end
 
+  def searches
+    Pipeline::Search.new(pipeline: self)
```
Comment (jlocke2) [inline lib/pipeline.rb:75]: Should `Pipeline::Search` be plural?

File: lib/pipeline/collection.rb [hunk 3]
```diff
@@ -43,7 +43,7 @@ def each(per_page: nil)
     pages = nil
     while pages.nil? || page < pages
       response = read_page(page: page, per_page: per_page)
-      pages ||= response.is_a?(Array) || response[collection_name] ? 1 : response["pagination"]["pages"]
+      pages ||= response.is_a?(Array) || response[collection_name] || response["entries"] ? 1 : response["pagination"]["pages"]
```
Comment (jlocke2) [inline lib/pipeline/collection.rb:46]: I'm not sure I understand the original logic or the additional change.

It seems like we are saying if the response is an array or contains the key mapping to the `collection_name`, then assume there is only one page.  This doesn't apply when there is the `entries` key, right?  I'm pretty sure when we set `entries`, we typically set the `pagination` as well.

Wouldn't it be safer to do something like...
```
pages ||= response.is_a?(Hash) && response.dig(:pagination, :pages) || 1
```

**Review by jlocke2 — COMMENTED** [review]
Looks good overall!

Couple minor questions though...

File: lib/pipeline/collection.rb (hunk 3 above)
Comment (sevgibson) [reply lib/pipeline/collection.rb:46]: I ran into something odd with `product_lines`, and you're right -- my change wasn't correct, but it did work, somehow! I'll try reverting and see what happens...

Comment (sevgibson) [reply lib/pipeline/collection.rb:46]: https://github.com/PipelineDeals/pipeline_api/pull/21/files#diff-c9d1387711f78d40463ec42c409655e077d1535f88a174ef54684ef41784eb6bR63

File: lib/pipeline.rb (hunk 2 above)
Comment (sevgibson) [reply lib/pipeline.rb:75]: I failed to add specs for the non-admin endpoints!!

File: lib/pipeline/collection.rb (hunk 3 above)
Comment (sevgibson) [reply lib/pipeline/collection.rb]: I like your suggestion...

File: lib/pipeline.rb (hunk 2 above)
Comment (sevgibson) [reply lib/pipeline.rb]: Fixed and added index spec. Not sure create/delete/update is worth it for this endpoint right now...

File: lib/pipeline/collection.rb (hunk 3 above)
Comment (jlocke2) [reply lib/pipeline/collection.rb]: 🎉 Great!

File: lib/pipeline.rb (hunk 2 above)
Comment (jlocke2) [reply lib/pipeline.rb]: Sounds great!

**Review by jlocke2 — APPROVED** [review]
🎉 Nicely done!

**Pull request merged**